### PR TITLE
Fix project (added user access rights check)

### PR DESCRIPTION
**Pull request by RomanKlishch**
- changed type fields of id (to long);
- added field User user to Card (added to scripts)
- added user access rights check;
- changed implementation of methods deleteCardById and leaveByCardId;
- changed tests

### CODE REVIEW

File: src/test/java/com/greetingcard/web/servlet/card/CreateCardServletTest.java [hunk 1]
```diff
@@ -32,19 +32,19 @@
...
+//        when(request.getSession()).thenReturn(session);
+//        when(session.getAttribute("user")).thenReturn(user);
+//
+//        servlet.doPost(request,response);
+//
+//        verify(request).getParameter("create-card");
+//        verify(cardService).createCard(card);
+//        verify(response).sendRedirect("/editCard");
+//    }
 }
```
Comment (OleksandrGubariev) [inline src/test/java/com/greetingcard/web/servlet/card/CreateCardServletTest.java:50]: Why this code is commented? You shouldn't add the commented code to develop

File: src/test/java/com/greetingcard/dao/jdbc/mapper/CardRowMapperTest.java [hunk 2]
```diff
@@ -25,23 +26,43 @@
...
+        assertEquals("lastName" , actualUser.getLastName());
+        assertEquals("login" , actualUser.getLogin());
+        assertEquals("email" , actualUser.getEmail());
+
         assertEquals(1, actualCard.getId());
-        assertEquals("Card", actualCard.getName());
-        assertEquals("/link", actualCard.getBackgroundImage());
-        assertEquals("/link", actualCard.getCardLink());
+        assertEquals("Card" , actualCard.getName());
+        assertEquals("/link" , actualCard.getBackgroundImage());
```
Comment (OleksandrGubariev) [inline src/test/java/com/greetingcard/dao/jdbc/mapper/CardRowMapperTest.java:64]: Why did you add spaces?

File: src/test/java/com/greetingcard/dao/jdbc/JdbcCongratulationDaoITest.java [hunk 3]
```diff
@@ -148,25 +147,25 @@ void saveLinksTest() throws SQLException {
         //then
         Congratulation congratulation = jdbcCongratulationDao.getCongratulationById(6);
         Link actualLink = congratulation.getLinkList().get(0);
-        assertEquals("you_tube_3", actualLink.getLink());
+        assertEquals("you_tube_3" , actualLink.getLink());
```
Comment (OleksandrGubariev) [inline src/test/java/com/greetingcard/dao/jdbc/JdbcCongratulationDaoITest.java:150]: ??

**Review by AlexKievDeveloper — APPROVED** [review]

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java [hunk 4]
```diff
@@ -32,64 +40,74 @@ public Congratulation getCongratulationById(int congratulationId) {
...
+                        try {
+                            Files.deleteIfExists(Paths.get(pathToFiles, file));
+                        } catch (IOException e) {
+                            log.error("Exception while deleting file - {}{}" , pathToFiles, file, e);
+                            throw new RuntimeException("Exception while deleting file" , e);
+                        }
+                    }
+                }
+            }
             statement.execute();
```
Comment (OleksandrGubariev) [inline src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java:106]: We remove all congratulation from card and we remove all files for congratulation. Do we remove the links for this congratulation from DB?

**Review by OleksandrGubariev — APPROVED** [review]
